tx.go

```diff
@@ -561,10 +561,13 @@ func (tx *Tx) writeMeta() error {
 	tx.meta.Write(p)
 
 	// Write the meta page to file.
+	tx.db.metalock.Lock()
```

Comment:
There might be some performance penalty under high concurrent readonly TXNs (especially short live TXNs)